apps/server/src/modules/tool/uc/external-tool-configuration.uc.ts

```diff
@@ -56,4 +141,20 @@ export class ExternalToolConfigurationUc {
 			requiredPermissions: [Permission.SCHOOL_TOOL_ADMIN],
 		});
 	}
+
+	private async ensureContextPermission(
+		userId: EntityId,
+		contextId: EntityId,
+		contextType: ToolContextType
+	): Promise<void> {
+		return this.authorizationService.checkPermissionByReferences(
```

Comment:
there also is a code smell on here which you might wana check:
https://sonarcloud.io/project/issues?resolved=false&types=CODE_SMELL&pullRequest=4144&id=hpi-schul-cloud_schulcloud-server&open=AYhNv7Qv8vu86PP2T-rb

Reply:
It says deprecated, but there is no alternative method mentioned in the doc that should be used instead.